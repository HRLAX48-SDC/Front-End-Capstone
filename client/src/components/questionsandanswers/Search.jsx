import React, {useState} from 'react';
import { Form, InputGroup } from 'react-bootstrap';

const Search = () => {
  const [searchTerm, setSearchTerm] = useState('');
  const getSearchTerm = (e) => {
    if(e.target.value.length > 3) {
      console.log(e.target.value);
    }
  }
  return (
    <Form className='q_searchbar' fluid>
      <InputGroup>
      <i class="fas fa-search"></i>
        <Form.Control
          type='search'
          placeholder='HAVING A QUESTION? SEARCH FOR ANSWERS...'
          onChange={getSearchTerm}
          // onKeyUp={getSearchTerm}
        >
        </Form.Control>
        <InputGroup.Text><i class="fas fa-search"></i></InputGroup.Text>
      </InputGroup>
    </Form>
  )
}
export default Search;