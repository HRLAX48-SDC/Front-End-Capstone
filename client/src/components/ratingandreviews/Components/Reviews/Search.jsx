import React, {useState, useEfect} from 'react';
import { Form, InputGroup } from 'react-bootstrap';

const Search = ({search, setSearch}) => {
  return (
    <Form>
        <InputGroup>
        <InputGroup.Text>
        <i className="fas fa-search"></i>
        </InputGroup.Text>
        
        <Form.Control
            type="text"
            placeholder="Search reviews..."
            value={search} 
            onChange={(e) => setSearch(e.target.value)}
            />
        </InputGroup>
    </Form>
  )
}
export default Search;