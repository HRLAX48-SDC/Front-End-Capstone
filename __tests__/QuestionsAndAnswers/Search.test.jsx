import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Search from '../../client/src/components/questionsandanswers/Search.jsx';
import { Button, Modal, Form, FloatingLabel} from 'react-bootstrap';
Enzyme.configure({ adapter: new Adapter()});

describe('<Search />', () => {
  const onChangeMock = jest.fn();
  test('render correctly', () => {
    const search = renderer.create(<Search searchTerm={'searchTerm'} setSearchTerm={onChangeMock}/>).toJSON();
    expect(search).toMatchSnapshot();
  });
  test('test onChange', () => {
    const wrapper = shallow(<Search searchTerm={'searchTerm'} setSearchTerm={onChangeMock}/>);
    wrapper.find('#q_searchinput').simulate('change', { target: {value: 'test'}});
    expect(onChangeMock.mock.calls.length).toBe(1);
  });
});