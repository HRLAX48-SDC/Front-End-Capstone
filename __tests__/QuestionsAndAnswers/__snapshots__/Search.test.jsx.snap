// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> render correctly 1`] = `
<form
  className=""
>
  <input
    className="form-control"
    id="q_searchinput"
    onChange={[Function]}
    placeholder="ï€‚ HAVING A QUESTION? SEARCH FOR ANSWERS..."
    type="text"
    value="searchTerm"
  />
</form>
`;
